### PR TITLE
Changes for integration into Habitat-Lab

### PR DESCRIPTION
**Pull request by exhaustin**
## Changelog

- Remove sophuspy so as to not introduce a hard-to-install dependency into habitat-lab
- Change configuration API for the goto controller to directly accept DictConfigs
- Add a relative goal interface for the goto controller

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/home_robot/home_robot/control/goto_controller.py [hunk 1]
```diff
@@ -4,19 +4,48 @@
...
 
 from .feedback.velocity_controllers import DDVelocityControlNoplan
 
 
 log = logging.getLogger(__name__)
 
+DEFAULT_CFG_NAME = "noplan_velocity_sim"
+
+
+def xyt_global_to_base(xyt_world2target, xyt_world2base):
```
Comment (cpaxton) [inline src/home_robot/home_robot/control/goto_controller.py:21]: should we add comments/doc for this kind of thing?

File: src/home_robot/home_robot/control/goto_controller.py [hunk 2]
```diff
@@ -4,19 +4,48 @@
...
+
+    xyt_base2target = np.zeros(3)
+    xyt_base2target[0] = x_diff * base_cos + y_diff * base_sin
+    xyt_base2target[1] = x_diff * -base_sin + y_diff * base_cos
+    xyt_base2target[2] = theta_diff
+
+    return xyt_base2target
+
+
+def xyt_base_to_global(xyt_base2target, xyt_world2base):
```
Comment (cpaxton) [inline src/home_robot/home_robot/control/goto_controller.py:36]: same

File: src/home_robot/home_robot/control/goto_controller.py [hunk 3]
```diff
@@ -26,11 +55,10 @@ class GotoVelocityController:
 
     def __init__(
         self,
-        cfg_name: Optional[str] = None,
+        cfg: Optional["DictConfig"] = None,
```
Comment (cpaxton) [inline src/home_robot/home_robot/control/goto_controller.py:58]: why?

**Review by cpaxton — APPROVED** [review]
lgtm, should document better though

File: src/home_robot/home_robot/control/goto_controller.py (hunk 3 above)
Comment (exhaustin) [reply src/home_robot/home_robot/control/goto_controller.py:58]: Habitat-lab uses hydra for configuration.
Passing in cfg_name uses the home_robot configuration system, which currently is a minimal placeholder I created.
Changing the input to a DictConfig will allow for config propagation from Habitat-Lab

**Pull request merged**